File: app/internal/publisher/template/go/device_replace.go
package wsd

import (
	"fmt"
	"strings"
)

type DeviceReplaceDialog struct {
	engine     *LicenseEngine
	licenseKey string
}

func NewDeviceReplaceDialog(engine *LicenseEngine, licenseKey string) *DeviceReplaceDialog {
	return &DeviceReplaceDialog{
		engine:     engine,
		licenseKey: licenseKey,
	}
}

func (d *DeviceReplaceDialog) Show() map[string]interface{} {
	fmt.Println(strings.Repeat("=", 60))
	fmt.Println("REPLACE DEVICE")
	fmt.Println(strings.Repeat("=", 60))
	fmt.Println()
	fmt.Println("Device reactivation requires Websmith Support approval.")
	fmt.Println()
	supportEmail := "support@websmithdigital.com"
	if d.engine != nil && d.engine.config != nil {
		if b, ok := d.engine.config["branding"].(map[string]interface{}); ok {
			if e, ok := b["support_email"].(string); ok && e != "" {
				supportEmail = e
			}
		}
	}
	fmt.Printf("Please contact support at: %s\n", supportEmail)
	fmt.Println("The application will remain locked until reactivation is approved.")
	fmt.Println()
	return map[string]interface{}{"action": "contact_support", "support_email": supportEmail}
}

File: app/internal/publisher/template/rust/src/device_replace.rs
use crate::license_engine::LicenseEngine;

pub struct DeviceReplaceDialog;

impl DeviceReplaceDialog {
    pub fn show(engine: &mut LicenseEngine, _license_key: &str) -> Result<DeviceReplaceResult, String> {
        let support_email = engine.config.as_ref()
            .and_then(|c| c.get("branding"))
            .and_then(|b| b.get("support_email"))
            .and_then(|v| v.as_str())
            .unwrap_or("support@websmithdigital.com")
            .to_string();

        println!("{}", "=".repeat(60));
        println!("REPLACE DEVICE");
        println!("{}", "=".repeat(60));
        println!();
        println!("Device reactivation requires Websmith Support approval.");
        println!();
        println!("Please contact support at: {}", support_email);
        println!("The application will remain locked until reactivation is approved.");
        println!();

        Ok(DeviceReplaceResult {
            action: "contact_support".to_string(),
        })
    }
}

pub struct DeviceReplaceResult {
    pub action: String,
}
